sql/executor.go

```diff
@@ -429,6 +439,56 @@ func (p parameters) Arg(name string) (parser.Datum, bool) {
 	}
 }
 
+var _ parser.Args = golangParameters{}
+
+type golangParameters []interface{}
```

Comment:
instead of `[]interface{}`, could you construct the `parser.Datum` at the caller and make this `[]parser.Datum`?